crates/api_crud/src/post/create.rs

```diff
@@ -120,9 +121,12 @@ pub async fn create_post(
     )
   };
 
+  plugin_hook_before("before_create_local_post", &mut post_form).await?;
```

Comment:
Again I think its much safer and more traceable if these returned forms, rather than alter them. Exterior mutability is not a good idea.

Reply:
It could be written as `post_form = plugin_hook_before("before_create_local_post", &post_form).await?;` but that would write exactly the same data to the same variable, and it would be a bit more verbose.

Reply:
I like that better; I'd rather it be verbose and traceable.

Reply:
Alright changed.